src/ape/_cli/__init__.py

```diff
@@ -0,0 +1,32 @@
+import click
+from ape_compile._cli import cli as compile
+from ape_console._cli import cli as console
+from ape_networks._cli import cli as networks
+from ape_plugins._cli import cli as plugins
+
+from ape._cli.groups import ApeCLI
+from ape._cli.options import shared_options
+from ape_accounts._cli import cli as accounts  # type: ignore
+
+CONTEXT_SETTINGS = {
+    "help_option_names": ["-h", "--help"],
+    "max_content_width": 200,
+}
+
+
+@click.group(
+    cls=ApeCLI,
+    context_settings=CONTEXT_SETTINGS,
+    invoke_without_command=True,
+    no_args_is_help=True,
+)
+@shared_options()
```

Comment:
This should be the only place that these are used (`ape --version` at least)

I'm open to adding `ape_config` for first displaying the config and then potentially working with it too

Reply:
What I would like to do is create a custom context object that is attached the CLI via a stateful object that gets set up automatically by reading the file.

Then, in order to use that config object, you'd have to re-use this `@shared_options` and it will give you another param. It won't matter that it's also here.

I have done this same pattern in a few other click projects to handle automatically creating an HTTP client for a particular service. I have also done a JSON config type situation.

Reply:
ah gotcha, okay I'll keep my eye out for that

Could we at least split them into the `version_option` and the `config_option`?